stl/inc/sstream

```diff
@@ -67,23 +67,88 @@ public:
+    basic_stringbuf& operator=(basic_stringbuf&& _Right) noexcept(
+        _Choose_pocma_v<_Alloc> != _Pocma_values::_No_propagate_allocators) /* strengthened */ {
+        if (this != _STD addressof(_Right)) {
+            _Assign_rv_no_alias(_STD move(_Right));
+        }
         return *this;
     }
 
-    void _Assign_rv(basic_stringbuf&& _Right) noexcept {
-        if (this != _STD addressof(_Right)) {
-            _Tidy();
-            this->swap(_Right);
+    void _Take_contents(basic_stringbuf&& _Right) noexcept {
+        // pre: *this holds no dynamic buffer and _Al == _Right._Al
+        _Mystate  = _STD exchange(_Right._Mystate, 0);
+        _Seekhigh = _STD exchange(_Right._Seekhigh, nullptr);
+        _Mysb::swap(_Right);
+    }
+
+    void _Copy_into_self_and_tidy(basic_stringbuf&& _Right) {
+        // pre: *this holds no dynamic buffer and _Al != _Right._Al
+        if ((_Right._Mystate & _Allocated) == 0) {
+            return;
+        }
+
+        const auto _Right_data     = _Right._Mysb::eback();
+        const auto _Right_capacity = static_cast<typename allocator_traits<_Alloc>::size_type>(
+            (_Right._Mysb::pptr() ? _Right._Mysb::epptr() : _Right._Mysb::egptr()) - _Right_data);
+
+        auto _New_capacity   = _Right_capacity;
+        const auto _New_data = _STD _Unfancy(_STD _Allocate_at_least_helper(_Al, _New_capacity));
+
+        _Mysb::setg(_New_data, _New_data + (_Right._Mysb::gptr() - _Right_data),
+            _New_data + (_Right._Mysb::egptr() - _Right_data));
+        if (_Right._Mysb::pbase() != nullptr) {
+            const auto _Pbase_diff = _Right._Mysb::pbase() - _Right_data;
+            const auto _Pptr_diff  = _Right._Mysb::pptr() - _Right_data;
+            const auto _Epptr_diff = _Right._Mysb::epptr() - _Right_data;
+            _Mysb::setp(_New_data + _Pbase_diff, _New_data + _Pptr_diff, _New_data + _Epptr_diff);
+        } else {
+            _Mysb::setp(nullptr, nullptr, nullptr);
+        }
+
+        const auto _Right_view = _Right._Get_buffer_view();
+        if (_Right_view._Ptr != nullptr) {
+            _Traits::copy(_New_data + (_Right_view._Ptr - _Right_data), _Right_view._Ptr, _Right_view._Size);
+        }
+
+        _Seekhigh = _New_data + _New_capacity;
+        _Mystate  = _Right._Mystate;
+
+        _Right._Tidy();
+    }
+
+    void _Assign_rv_no_alias(basic_stringbuf&& _Right) noexcept(
+        _Choose_pocma_v<_Alloc> != _Pocma_values::_No_propagate_allocators) {
+        // pre: this != std::addressof(_Right)
+        _Tidy();
+        if constexpr (_Choose_pocma_v<_Alloc> == _Pocma_values::_No_propagate_allocators) {
+            if (_Al == _Right._Al) {
+                _Take_contents(_STD move(_Right));
+            } else {
+                _Copy_into_self_and_tidy(_STD move(_Right));
+            }
+        } else {
+            if constexpr (_Choose_pocma_v<_Alloc> == _Pocma_values::_Propagate_allocators) {
+                _Al = _STD move(_Right._Al);

```

Comment:
This line was miswritten as `_Al = _STD move(_Al);` (931310e812060664fe17daf9a863dbc39f51f855) but it seemed that no test, including the new version of [`std/input.output/string.streams/stringbuf/stringbuf.cons/move.alloc.pass.cpp`](https://github.com/llvm/llvm-project/blob/185302530847a28f44e48a67a79fd4eba048a1c7/libcxx/test/std/input.output/string.streams/stringbuf/stringbuf.cons/move.alloc.pass.cpp), was affected. Perhaps we should add test coverage for POCMA.